rust/src/delta.rs

```diff
@@ -707,6 +706,69 @@ impl DeltaTable {
             .collect::<Vec<String>>())
     }
 
+    /// Whether a path should be hidden for delta-related file operations, such as Vacuum.
+    /// Names of the form partitionCol=[value] are partition directories, and should be
+    /// deleted even if they'd normally be hidden. The _db_index directory contains (bloom filter)
+    /// indexes and these must be deleted when the data they are tied to is deleted.
+    fn is_hidden_directory(&self, path_name: &str) -> Result<bool, DeltaTableError> {
```

Comment:
The lack of unit tests here make me uneasy, considering the importance not deleting data that shouldn't be deleted :laughing: 

For the purposes of vacuum is it insufficient to just delete files ending in `.parquet` :thinking:

Reply:
I agree! We should add integration tests to validate the different scenarios.